lib/elixir/lib/exception.ex

```diff
@@ -787,7 +787,14 @@ defmodule File.Error do
   defexception [:reason, :path, action: ""]
 
   def message(exception) do
-    formatted = IO.iodata_to_binary(:file.format_error(exception.reason))
+    formatted =
+      case exception.reason do
+        :eexist ->
+          "directory is not empty"
```

Comment:
We should change this exclusively inside `rmdir!` as the "eexist" reason is valid in other situations. For example:

```iex
iex> File.mkdir "foo"
:ok
iex> File.mkdir "foo"
{:error, :eexist}
```